redash/query_runner/pg.py

```diff
@@ -63,6 +63,38 @@ def _wait(conn, timeout=None):
             raise psycopg2.OperationalError("select.error received")
 
 
+def full_table_name(schema, name):
+    if '.' in name: 
+        name = u'"{}"'.format(name)
+    
+    return u'{}.{}'.format(schema, name)
+
+
+def build_schema(query_result, schema):
+    # By default we omit the public schema name from the table name. But there are
+    # edge cases, where this might cause conflicts. For example:
+    # * We have a schema named "main" with table "users".
+    # * We have a table named "main.users" in the public schema.
+    # (while this feels unlikely, this actually happened)
+    # In this case if we omit the schema name for the public table, we will have
+    # a conflict.
+    table_names = set(map(lambda r: full_table_name(r['table_schema'], r['table_name']), query_result['rows']))
+
+    for row in query_result['rows']:
+        if row['table_schema'] != 'public':
+            table_name = full_table_name(row['table_schema'], row['table_name'])
+        else:
+            if row['table_name'] in table_names:
+                table_name = full_table_name(row['table_schema'], row['table_name'])
+            else:
+                table_name = row['table_name']
```

Comment:
Since this is an edge case, maybe it'll be best if it has less distraction?

```suggestion
        table_name = row['table_name']
                
        if row['table_schema'] != 'public' or table_name in table_names:
            table_name = full_table_name(row['table_schema'], row['table_name'])                
```

Reply:
I remember having some issue with doing it this way, but looking at it now it seems legit. :man_shrugging: